clang/test/Parser/old-style-declaration.c

```diff
@@ -0,0 +1,33 @@
+// RUN: %clang_cc1 -fsyntax-only -verify -Wold-style-declaration %s
+// RUN: %clang_cc1 -fsyntax-only -verify -Wextra %s
+
```

Comment:
It would be good to add a RUN line along these lines:
```suggestion
// RUN: %clang_cc1 -fsyntax-only -verify -Wextra %s
// RUN: %clang_cc1 -fsyntax-only -verify=cpp -Wextra -x c++ %s

// cpp-no-diagnostics
```
and group the code that's invalid in C++ under `#ifndef __cplusplus`, to show that the diagnostics are not generated in C++, only in C.